src/openrct2/management/finance.c

```diff
@@ -125,17 +125,19 @@ void finance_pay_research()
  */
 void finance_pay_interest()
 {
-    money32 current_loan = gBankLoan;
-    uint8 current_interest = gBankLoanInterestRate;
+	// This variable uses the 64-bit type as the computation below can involve multiplying very large numbers
+    // that will overflow money32 if the loan is greater than (1 << 31) / (5 * current_interest_rate)
+	money64 current_loan = gBankLoan;
+    uint8 current_interest_rate = gBankLoanInterestRate;
+	money32 interest_to_pay;
```

Comment:
Shouldn't this be `money64` too, as you're storing the result of multiplying a `money64` with other values in it?